src/realm/object-store/c_api/app.cpp

```diff
@@ -253,6 +253,24 @@ RLM_API realm_app_t* realm_app_create(const realm_app_config_t* app_config,
     });
 }
 
+RLM_API realm_app_t* realm_app_create_cached(const realm_app_config_t* app_config,
+                                             const realm_sync_client_config_t* sync_client_config)
+{
+    return wrap_err([&] {
+        return new realm_app_t(App::get_shared_app(*app_config, *sync_client_config));
+    });
+}
+
+RLM_API realm_app_t* realm_app_get_cached(const char* app_id, const char* base_url)
+{
+    if (auto app =
```

Comment:
just a NIT, I would use  `wrap_err` here as well, although I am not sure whether `App::get_cached_app` can throw, `new` and `std::string `constructor can definitely throw in some corner case situation (when memory is low).

Reply:
Hm... so then I guess I'll then need to change the return type to `bool` and move the app to an argument?

Reply:
Not necessarily `wrap_err`would return `nullptr` in case of error, but yes, it always the same problem with the C-API, we need to signal the error somehow.  You can either assume that `nullptr` in this case means error, provided `App::get_cached_app` does not throw.

Reply:
I think the issue here would be that `nullptr` is a valid return value - in case you're asking for an app that wasn't previously cached, which would mean there's no way to distinguish between `nullptr` returned due to the app not being cached and `nullptr` being returned due to an error in which case you need to go and fetch last error and throw it.

Reply:
indeed.. yes. You are right.

Reply:
Thanks, I changed it to return a bool value for success and the app as an out parameter based on the patterns used elsewhere in the file. Seems to work fine to me, but would appreciate a quick sanity check to confirm I'm not missing something.